gqcp/include/Mathematical/Representation/Tensor.hpp

```diff
@@ -423,17 +423,37 @@ class Tensor:
 
         GQCP::Matrix<Scalar> M {rows, cols};
 
-        for (int i = 0; i < this->dimension(0); i++) {
-            for (int j = 0; j < this->dimension(1); j++) {
-                for (int k = 0; k < this->dimension(2); k++) {
-                    for (int l = 0; l < this->dimension(3); l++) {
-                        const auto row_index = i + j + i * (this->dimension(1) - 1);
-                        const auto col_index = k + l + k * (this->dimension(3) - 1);
-
-                        M(row_index, col_index) = this->operator()(i, j, k, l);
+        if (rows == cols) {
+            for (int i = 0; i < this->dimension(0); i++) {
+                for (int j = 0; j < this->dimension(1); j++) {
+                    for (int k = 0; k < this->dimension(2); k++) {
+                        for (int l = 0; l < this->dimension(3); l++) {
+                            const auto row_index = i + j + i * (this->dimension(1) - 1);
```

Comment:
Ah yes, I believe you're right. But Since I don't have to split up the implementation, this will probably be removed entirely, as the other case is more general.